### ISSUE TITLE
req.body gives me undefined often

### ISSUE THREAD

**Issue by mgenev** [balderdashy/sails#2497]
And for no apparent reason at all. Some requests go through, others do not....

On the server I have this line:
```
var record = req.body.record;
var file = req.file('upload');
```

and consistently from the client I'm sending a form with 'record' in it. I also have the file coming through this request:

```
------WebKitFormBoundaryblqLPpqy5AsUxBwK
Content-Disposition: form-data; name="upload"; filename="1.png"
Content-Type: image/png


------WebKitFormBoundaryblqLPpqy5AsUxBwK
Content-Disposition: form-data; name="record"

{"template":{"title":"fasfasfsa"}}
------WebKitFormBoundaryblqLPpqy5AsUxBwK--
```

I have no clue why sometimes the server fails and sometimes it doesn't...

**Comment by corbanb** [comment]
+1 this is all the time for me right now.

**Comment by CWyrtzen** [comment]
Still an issue?

**Comment by jetsonjohn** [comment]
I have the same issue for inputs declared after file input type. The text input type values is empty in the request

**Comment by sailsbot** [comment]
Thanks for posting, @mgenev.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by wongzigii** [comment]
I am suffering this issue, why this one is being ignored?

**Comment by shekhardesigner** [comment]
`req.body` will be `undefined` if you removed `bodyParser` from http config. 
Make sure its added in middleware order in `sails.config.http` or `configs/http.js` file 

```
/**
 * (sails.config.http)
 * https://sailsjs.com/config/http
 */

module.exports.http = {
    middleware: {

        order: [
            'bodyParser',
            'poweredBy',
            ...
        ],
    },

};

```
